lib/Dialect/MSFT/MSFTOps.cpp

```diff
@@ -809,5 +810,93 @@ hw::ModulePortInfo MSFTModuleExternOp::getPorts() {
 
 void OutputOp::build(OpBuilder &builder, OperationState &result) {}
 
+//===----------------------------------------------------------------------===//
+// MSFT high level design constructs
+//===----------------------------------------------------------------------===//
+
+// let assemblyFormat = [{
+//   $rowInputs `:` type($rowInputs) `,`
+//   $colInputs `:` type($colInputs)
```

Comment:
Since this is here, another option I considered when looking at this was something like this:

```
$rowInputs, $colInputs attr-dict `:` function-type(operands(), results()) ...`
```

Just my two cents but I kind of prefer if the operands/results and their types fit into the assemblyFormat, since that makes it a little more uniform with other IRs that also use it.

Reply:
I started with something like that, but it was just too redundant. Too many `!hw.array<`s which could have been inferred. (Especially the pe results.)

Reply:
There was a relatively new feature to be able to elide those type prefixes... I guess that only works when the op and type are in the same dialect?

I'm still finding the current form a little suprising as I stare at the test. What about instead of `2 x %rowInputs : i8`, using `%rowInputs : 2 x i8`?

Reply:
this may all be for naught... possible that we'll want to break out the row/column inputs from hw.arrays to individual ports to simplify building the schedule dep graph.